### ISSUE TITLE
make setBatchGenerateT1map accept both SlicePartialFourier and PartialFourierInSlice as valid input metadata

### ISSUE THREAD

**Issue by Remi-Gau** [cpp-lln-lab/bidspm#1030]
add fall back to use the "bids" metadata name if the expected one is not found:

https://github.com/cpp-lln-lab/bidspm/blob/0c5309733f6c10abe7f43b566957174c3c2a203b/src/batches/preproc/setBatchGenerateT1map.m#L33

look for `PartialFourierInSlice` and if it fails try `SlicePartialFourier`, if it finds it mention this with a DEBUG log entry and if it fails then throw WARNING log

**Comment by JacMatu** [comment]
One thing that would be helpful to a (naive) user would be a more explicit warning/error message. 

First, when I tried to run the script I got explicit errors about expected JSON parameters that should be there for the MP2RAGE toolbox.


```
BUILDING JOB: generate t1 map
Reference to non-existent field 'RepetitionTimePreparation'.

Error in setBatchGenerateT1map (line 123)
      estimateT1.TR = metadataInv1.RepetitionTimePreparation;

Error in bidsGenerateT1map (line 24)
    matlabbatch = setBatchGenerateT1map(matlabbatch, BIDS, opt, subLabel);

Error in generateT1map (line 18)
matlabbatch = bidsGenerateT1map(opt);
```

And that was pretty explicit in pointing me to the source of error that something is missing from the JSON files. 

Then came the `SlicePartialFourier` error 

```
Warning: Missing non-BIDS metadata for
    - acq: {r0p75}
    - modality: {anat}
    - prefix: {}
    - space: {}
    - suffix: {MP2RAGE}
    - sub: {EB01}
    - ses: {01}
    - run: {}
    - inv: {2}
See 'help setBatchGenerateT1map' 
> In errorHandling (line 56)
  In setBatchGenerateT1map (line 150)
  In bidsGenerateT1map (line 24)
  In generateT1map (line 21) 
Warning: This batch is empty & will not be run. 
> In errorHandling (line 56)
  In saveAndRunWorkflow (line 40)
  In bidsGenerateT1map (line 26)
  In generateT1map (line 21) 
Index exceeds the number of array elements (0).

Error in bidsGenerateT1map (line 34)
    prefix = matlabbatch{1}.spm.tools.mp2rage.estimateT1.outputT1.prefix;

Error in generateT1map (line 21)
matlabbatch = bidsGenerateT1map(opt);
```

And with this one  I had no idea what's the problem and how to solve it. It's weird that the errors were different in these cases, isn't it? In both of them, MP2RAGE toolbox expects specific fields in JSON but cannot find it (either it's totally missing or it's not named as expected). 

TL;DR: I think it would be handy if the errors/warnings were more explicit here (just like in the case of missing fields from JSONS)

**Comment by Remi-Gau** [comment]
yeah ideally I want to the errors / warning to say what info was expected in what files, what was found or not in these files instead and ideally pointing to resources how how to fix the issue (ideally in the doc or the FAQ or some other resource).

**Comment by Remi-Gau** [comment]
> the json file for the MP2RAGE comes from heudiconv conversion.

yeah but heudiconv takes most of the json from dicom headers like other dicom converters and I doubt that any tool at the moment is able to "find" the details that are needed for T1map generation.

> Just to clarify, what is not BIDS friendly? The json file of the MP2RAGE or the toolbox?

Well `PartialFourierInSlice` appears no where in the spec. But `RepetitionTimePreparation` that you have to add into the json is actually a BIDS metadata.

So it would make more sense to stick to all BIDS terms and so that `SlicePartialFourier` can be used even if not an official BIDS term (not official only honorable mention)
